File: front-end/src/types/ExerciseResultDTO.ts
export interface ExerciseResultDTO {
  attemptId: number;
  score: number;
  percentage: number;
  passed: boolean;
  timeSpent?: number;
  totalQuestions: number;
  correctAnswers: number;
  incorrectAnswers: number;
  unansweredQuestions: number;
  teacherFeedback?: string;
  submittedAt?: string;
  gradedAt?: string;
}

File: back-end/src/main/java/com/meutcc/backend/content/exercise/dtos/ExerciseResultDTO.java
package com.meutcc.backend.content.exercise.dtos;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.meutcc.backend.content.question.dtos.QuestionResultDTO;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@JsonInclude(JsonInclude.Include.NON_NULL)
public record ExerciseResultDTO(
        Long attemptId,
        BigDecimal score,
        BigDecimal percentage,
        Boolean passed,
        Integer timeSpent,
        Integer totalQuestions,
        Integer correctAnswers,
        Integer incorrectAnswers,
        Integer unansweredQuestions,
        List<QuestionResultDTO> questionResults,
        String teacherFeedback,
        LocalDateTime submittedAt,
        LocalDateTime gradedAt
) {}